src/app/dialogs/charts/farm-buybacks-dialog/farm-buybacks-dialog.component.ts

```diff
@@ -21,37 +23,50 @@ export class FarmBuybacksDialogComponent extends ChartGeneralMethodsComponent im
               private hardworksService: HardworksService,
               private priceData: PriceDataService,
               private tvlsService: TvlsService,
+              private rewardsService: RewardsService,
   ) {
     super(cdRef, vt);
   }
 
   load(): void {
-    this.hardworksService.getHardWorkHistoryData(StaticValues.NETWORKS.get(this.network)).subscribe(data => {
-      this.log.debug('History of All Farm buybacks loaded ', data);
-      const chartBuilder = new ChartBuilder();
-      chartBuilder.initVariables(2);
-      data?.forEach(dto => {
-        let bb = dto.farmBuybackSum / 1000;
-        if (dto.network === 'bsc') {
-          const farmPrice = this.priceData.getLastFarmPrice();
-          if (farmPrice && farmPrice !== 0) {
-            bb = bb / farmPrice;
-          } else {
-            bb = 0;
-          }
-        }
-        chartBuilder.addInData(0, dto.blockDate, bb);
-      });
+    const rewardsRequest = this.rewardsService.getHistoryRewards('PS', StaticValues.NETWORKS.get(this.network));
+    const hardworkRequest = this.hardworksService.getHardWorkHistoryData(StaticValues.NETWORKS.get(this.network));
+    const historyTvlRequest = this.tvlsService.getHistoryTvlByVault('PS');
+
+    forkJoin([rewardsRequest, hardworkRequest, historyTvlRequest])
+      .subscribe(([rewards, data, vaultData]) => {
+        this.log.debug('History of All rewards ', rewards);
+        this.log.debug('History of All Farm buybacks loaded ', data);
+        this.log.debug('History of PS TVL loaded ', vaultData);
+
+        const totalReward = rewards
```

Comment:
need to compare farmBuyBack with sum of rewards emitted on the same block
it is a bit more complex computation
isWeeklyRewards will be updated for all DTOs soon